### ISSUE TITLE
Failed to run process: {cmd}

### ISSUE THREAD

**Issue by CobreDev** [AsahiLinux/asahi-installer#29]
**My hardware:**

Base 2020 M1 MacBook Air
- 8GB RAM
- 256GB SSD, roughly 112GB free

**My issue:**

When trying to run the Asahi installer script (running `curl https://alx.sh | sh` in terminal), it fails to resize my disk partition. Here are my logs:

https://0bin.net/paste/fXoXtV6a#O9LRFJ-VlF/LQepMyk5Z1arme1cEOlm3xkBiUAEvOze

**Comment by marcan** [comment]
That means your filesystem is corrupted; you'd have the same problem resizing it from the macOS Disk Utility. You need to run a repair and see if you can fix it.

Unfortunately, APFS doesn't have the best track record of stability or reliability... nothing we can do about that, unfortunately :(

**Comment by maximalisms** [comment]
> That means your filesystem is corrupted; you'd have the same problem resizing it from the macOS Disk Utility. You need to run a repair and see if you can fix it.
> 
> Unfortunately, APFS doesn't have the best track record of stability or reliability... nothing we can do about that, unfortunately :(

Rebuilt the APFS volume and this fixed the same issue OP was having. Had to wipe the whole drive.

**Comment by CobreDev** [comment]
I went into 1TR and ran First Aid, now whenever I try to run the installer it says this:

```Enter the new size for your existing partition:
  You can enter a size such as '1GB', a fraction such as '50%',
  or the word 'min' for the smallest allowable size.

  Examples:
  30%  - 30% to macOS, 70% to the new OS
  80GB - 80GB to macOS, the rest to your new OS
  min  - Shrink macOS as much as (safely) possible

» New size (min): 200GB

Resizing will free up 45.11 GB of space.

Note: your system may appear to freeze during the resize.
This is normal, just wait until the process completes.
» Continue? (y/N): y

Started APFS operation
Aligning shrink delta to 45,108,101,120 bytes and targeting a new physical store size of 199,999,094,784 bytes
Determined the minimum size for the targeted physical store of this APFS Container to be 245,107,195,904 bytes
Error: -69521: Your APFS Container resize request is below the APFS-system-imposed minimal container size (perhaps caused by APFS Snapshot usage by Time Machine)
Failed to run process: diskutil apfs resizeContainer disk0s2 199999094784
root        : ERROR    Process execution failed
Traceback (most recent call last):
  File "/private/tmp/asahi-install/main.py", line 806, in <module>
    InstallerMain().main()
  File "/private/tmp/asahi-install/main.py", line 658, in main
    while self.main_loop():
  File "/private/tmp/asahi-install/main.py", line 778, in main_loop
    return self.action_resize(parts_resizable)
  File "/private/tmp/asahi-install/main.py", line 598, in action_resize
    self.dutil.resizeContainer(target.name, val)
  File "/private/tmp/asahi-install/diskutil.py", line 191, in resizeContainer
    self.action("apfs", "resizeContainer", name, size, verbose=2)
  File "/private/tmp/asahi-install/diskutil.py", line 37, in action
    subprocess.run(["diskutil"] + list(args), check=True)
  File "/private/tmp/asahi-install/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['diskutil', 'apfs', 'resizeContainer', 'disk0s2', '199999094784']' returned non-zero exit status 1.```

**Comment by maximalisms** [comment]
> Edit: I need to read lol, I'll try messing with Time Machine Snapshots

If that fails, I would backup all the important information on your PC to a separate drive and just re-partition. This is what I had to do in order for the install to work properly.

**Comment by CobreDev** [comment]
> > Edit: I need to read lol, I'll try messing with Time Machine Snapshots
> 
> 
> 
> If that fails, I would backup all the important information on your PC to a separate drive and just re-partition. This is what I had to do in order for the install to work properly.

The best way to do that is just wipe the entire disk and start over, or what?

**Comment by maximalisms** [comment]
> > > Edit: I need to read lol, I'll try messing with Time Machine Snapshots
> > 
> > 
> > If that fails, I would backup all the important information on your PC to a separate drive and just re-partition. This is what I had to do in order for the install to work properly.
> 
> The best way to do that is just wipe the entire disk and start over, or what?

You will need to boot into startup options, then go to disk utility and erase your disk. Then you re-partition your disk with APFS and reinstall your MacOS. Once that is done, run the Asahi Installer again and it should work.

**Comment by marcan** [comment]
Duplicate of #81
